.github/workflows/ci.yml

```diff
@@ -869,6 +866,15 @@ jobs:
         run: |
           conda info
           conda list
+      - name: Start local ray cluster
+        run: |
+          ray start --head --port=6379 --object-store-memory=1000000000
+          while [ $? -ne 0 ]; do
```

Comment:
Did you find that you needed the retries in practice? I'm usually pretty wary about adding potential infinite loops here. Maybe like @vnlitvinov we should have some limit of retries. I think something like 10-20 is reasonable. Also, if the retries were needed, maybe we should do the same for `test-compat-windows`?

Reply:
I wonder if we can use stuff like https://github.com/marketplace/actions/retry-step to actually do the thing for us

Reply:
I switched to retry-step with a limited number of retries. `test-compat-windows` no longer exists. r.e. whether the retries help, see my experiment results in the PR description: https://github.com/modin-project/modin/pull/5101#issue-1398915504